cmd/git/branch.go

```diff
@@ -0,0 +1,76 @@
+package git
+
+import (
+  "fmt"
+  "io/ioutil"
+  "log"
+  "strings"
+
+  "github.com/Originate/gt/cmd/util"
+)
+
+
+func EnsureDoesNotHaveBranch(branchName string) {
+  if HasBranch(branchName) {
+    message := fmt.Sprintf("A branch named '%s' already exists", branchName)
+    util.ExitWithErrorMessage(message)
+  }
+}
+
+
+func GetCurrentBranchName() string {
+  if IsRebaseInProgress() {
+    filename := fmt.Sprintf("%s/.git/rebase-apply/head-name", GetRootDirectory())
+    content, err := ioutil.ReadFile(filename)
+    if err != nil {
+      log.Fatal(err)
+    }
+    return strings.Replace(strings.TrimSpace(string(content)), "refs/heads/", "", -1)
```

Comment:
Broke this into two lines as there isn't a good name for all that was going on here.